lib/archethic/contracts/interpreter/library/common/http_impl.ex

```diff
@@ -17,133 +17,134 @@ defmodule Archethic.Contracts.Interpreter.Library.Common.HttpImpl do
   @behaviour Http
   @threshold 256 * 1024
   @timeout Application.compile_env(:archethic, [__MODULE__, :timeout], 2_000)
+  @allow_http? Application.compile_env(:archethic, [__MODULE__, :allow_http?], false)
```

Comment:
Suggestion: Maybe you can use a list of supported schemes, this way it will be less specific to oriented https vs http but more generic, and I think we can support also later fetching from AEWeb/AEFS itself or other potentiel providers (ie. IPFS)

```elixir
@supported_schemes Application.compile_env(:archethic, [__MODULE__, :supported_schemes], [:https])
```

This will also simplify and optimize the check in `validate_scheme`

```elixir
defp validate_scheme(scheme) when scheme in @supported_schemes, do {:ok, scheme}
defp validate_scheme(_), do: {:error, :not_supported_scheme}
```

Reply:
Done in last commit